### DISCUSSION TITLE
AWS?

### DISCUSSION THREAD

**Discussion by ajmetz** [mojolicious/mojo#1921] [Q&A]
Any tried and tested ways to run Mojolicious Webapps via Amazon Web Services, people here have experience with? 

Never used AWS and suggested a Mojolicious project for an innovation week / hackathon at work, and a team was assembled, and on day one researched if using Amazon Web Services was viable. I'll hear the results of day one's research at the start of day two, =). Just wondered if this community had any two cents to offer?

**Comment by kraih** [comment]
Nothing special about running Mojolicious on EC2.

**Reply by ajmetz** [reply]
Thanks for the super quick response! =).
A quick google and I ended up here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EC2_GetStarted.html
I'd heard of different methods - Mojolicious in a docker container, Mojolicious in a linux instance, and Mojolicious on a serverless server via Lambda. Looking up EC2, I'm guessing it's similar to a Cloud VPS (I currently have a Cloud VPS via Neostrada / TotaalCloud with Ubuntu 21.10, Nginx and Perlbrew - would EC2 be the AWS equivalent then?).

**Comment by s1037989** [comment]
Agreed.  Nothing special.  Spin up an EC2 instance and run your Mojo app
like normal.  Open 80/443 in your AWS VPC Network ACLs and Security
Groups.  Ignore all the fancy AWS products -- all you need is EC2 (or
Lightsail).  Containers, S3, Route 53, databases...  all fine, but not
necessary.  You may grow your app to that, but do not start with such
complicated architecture for a hackathon.

**Comment by s1037989** [comment]
That is correct.  EC2 = VPS = VM.  When you spin up an Ubuntu EC2, it'll be
a VM like any other.
Yes, Mojolicious in Docker and Lambda are all fine and dandy, but that's
for scale.  Don't waste your precious hackathon time on infrastructure
architecture where you want to focus on your app.  Install a Postgres
database into your EC2 using `apt install` like anything else; I'd even
configure postgres to "trust" and not worry about authenticating to it.
Spin up an nginx application router if you really need one for your app,
otherwise just have Mojolicious listen on 80/443 directly.  Perlbrew is
great, but certainly not necessary.  Just use system perl and install
modules to system perl as root.  You might want to use Perlbrew to use the
latest greatest Perl, tho.  Of course, some of this is bad advice for
proper deployments...  but you're hacking.  Focus on your app and have
fun!  :)

**Comment by hbarnard** [comment]
```
[root@ip-172-30-0-27 cclite2.0.1-7-centos]# curl -L https://cpanmin.us | perl - -M https://cpan.metacpan.org -n Mojolicious
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  295k  100  295k    0     0  1275k      0 --:--:-- --:--:-- --:--:-- 1279k
Mojolicious is up to date. (9.23)
[root@ip-172-30-0-27 cclite2.0.1-7-centos]# morbo
bash: morbo: command not found
[root@ip-172-30-0-27 cclite2.0.1-7-centos]# hypnotoad
bash: hypnotoad: command not found
```
``

Part of the EC2 experience. I'm testing https://sourceforge.net/projects/cclite2/ on EC2, no problem for Ubuntu, but Centos seems problematic. I'm not using containers etc. the whole thing will run on a dedicated system later.

**Reply by daleif** [reply]
Isn't roots PATH always rather limited? So it depends on where the bins ended up being installed after the installation.

**Comment by hbarnard** [comment]
Hi Lars,

Yes, thanks. I've tried either way, root and non-root. What I want is 'easy' (I have a large install script) for non-expert. I do that with a deb for Ubuntu, Debian etc. I'll carry on investigating. The other thing is that there's less 'Perl' on Centos AWS, I have a lot of extra yum commands.

Best Hugh

---------
https://www.hughbarnard.org
Twitter: @hughbarnard

------- Original Message -------

**Reply by ajmetz** [reply]
cpanm -V shows a lot of the different paths, and you can check everything is installing to the correct place. That said, I only ever use Perlbrew for the most part. One day I ought to try plenv of course, ^_^.

**Reply by daleif** [reply]
DId you search for it? Just to see where morbo and hypnotaod went? (I offen use `locate morbo`)

**Comment by hbarnard** [comment]
Hi Lars

Thanks, I didn't and I've backed off now. Tried 'which' and it didn't find it. Really annoying because it's now almost seamless on Ubuntu. I'll take another run in a day or two and write about it again.

All the best Hugh

---------
https://www.hughbarnard.org
Twitter: @hughbarnard

------- Original Message -------

**Reply by daleif** [reply]
`which` just tells you it isn't in the PATH, you already knew that. That's why I suggested using `locate` instead. It searches a database of all the files on disk (updated via `sudo updatedb`) and thus can find programmes not in the PATH. Very useful.

**Comment by hbarnard** [comment]
AWS centos is /usr/local/bin/morbo and, of course hypnotoad...thanks to everyone for suggestions.

---------
https://www.hughbarnard.org
Twitter: @hughbarnard

------- Original Message -------
